app/components-react/root/LiveDock.tsx

```diff
@@ -0,0 +1,437 @@
+      if (this.restreamService.shouldGoLiveWithRestream) return 'restream';
+      return 'default';
+    }
+
+    return this.chatApps.find(app => {
+      if (app.id === this.store.underlyingSelectedChat) {
+        return app.id === this.store.underlyingSelectedChat;
+      }
+    })
+      ? this.store.underlyingSelectedChat
+      : 'default';
+  }
+
+  get applicationLoading() {
+    return this.appService.state.loading;
+  }
+
+  get streamingStatus() {
+    return this.streamingService.state.streamingStatus;
+  }
+
+  get isStreaming() {
+    return this.streamingService.isStreaming;
+  }
+
+  get collapsed() {
+    return this.customizationService.state.livedockCollapsed;
+  }
+
+  get liveText() {
+    if (this.streamingStatus === EStreamingState.Live) return 'Live';
+    if (this.streamingStatus === EStreamingState.Starting) return 'Starting';
+    if (this.streamingStatus === EStreamingState.Ending) return 'Ending';
+    if (this.streamingStatus === EStreamingState.Reconnecting) return 'Reconnecting';
+    return 'Offline';
+  }
+
+  get platform() {
+    return this.userService.platform?.type;
+  }
+
+  get offlineImageSrc() {
+    const mode = this.customizationService.isDarkTheme ? 'night' : 'day';
+    return require(`../../../media/images/sleeping-kevin-${mode}.png`);
+  }
+
+  get hideViewerCount() {
+    return this.customizationService.state.hideViewerCount;
+  }
+
+  get liveDockSize() {
+    return this.customizationService.state.livedockSize;
+  }
+
+  get viewerCount() {
+    if (this.hideViewerCount) {
+      return 'viewers hidden';
+    }
+    return this.streamingService.views.viewerCount.toString();
+  }
+
+  get hideStyleBlockers() {
+    return this.windowsService.state.main.hideStyleBlockers;
+  }
+
+  get hasChatTabs() {
+    return this.chatTabs.length > 1;
+  }
+
+  get showDefaultPlatformChat() {
+    return this.selectedChat === 'default';
+  }
+
+  get restreamChatUrl() {
+    return this.restreamService.chatUrl;
+  }
+
+  get chatApps(): ILoadedApp[] {
+    return this.platformAppsService.enabledApps.filter(app => {
+      return !!app.manifest.pages.find(page => {
+        return page.slot === EAppPageSlot.Chat;
+      });
+    });
+  }
+
+  get chatTabs(): { name: string; value: string }[] {
+    if (!this.userService.state.auth) return [];
+    const tabs: { name: string; value: string }[] = [
+      {
+        name: getPlatformService(this.userService.state.auth.primaryPlatform).displayName,
+        value: 'default',
+      },
+    ].concat(
+      this.chatApps
+        .filter(app => !app.poppedOutSlots.includes(this.store.slot))
+        .map(app => {
+          return {
+            name: app.manifest.name,
+            value: app.id,
+          };
+        }),
+    );
+    if (this.restreamService.shouldGoLiveWithRestream) {
+      tabs.push({
+        name: $t('Multistream'),
+        value: 'restream',
+      });
+    }
+    if (this.userService.state.auth.primaryPlatform === 'twitter') {
```

Comment:
Maybe we should add Instagram here as well, odds of that ending as primary platform are slim, but if we feel like being super safe.

Reply:
i'm going to leave this unchanged for now since that doesn't seem necessary currently, and will likely be less necessary after your changes